File: Selenium/Activities/activity9.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class activity9 {
    public static void main(String[] args) {
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));
        driver.get("https://www.training-support.net/selenium/ajax");
        System.out.println("The title of the page is: " + driver.getTitle());
        //Find and click the "Change content" button on the page.
        driver.findElement(By.xpath("//*[@id=\"ajax-content\"]/button")).click();
        //Wait for the text to say "HELLO!". Print the message that appears on the page.
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@id=\"ajax-content\"]/h1")));
        WebElement messageHello = driver.findElement(By.xpath("//*[@id=\"ajax-content\"]/h1"));
        System.out.println("The message on the page is: " + messageHello.getText());
        //Wait for the text to change to contain "I'm late!". Print the new message on the page.
        WebElement delayedText = driver.findElement(By.tagName("h3"));
        System.out.println(delayedText.getText());
        wait.until(ExpectedConditions.textToBePresentInElementLocated(By.xpath("//*[@id=\"ajax-content\"]/h3"),"I'm late!"));
        WebElement messageLate = driver.findElement(By.xpath("//*[@id=\"ajax-content\"]/h3"));
        System.out.println("The message on the page is: " + messageLate.getText());
        driver.quit();

    }
}

File: Selenium/Activities/Activity9.py
from selenium import webdriver
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.common.by import By

service = FirefoxService(GeckoDriverManager().install())
with webdriver.Firefox(service=service) as driver:
    driver.get("https://www.training-support.net/selenium/ajax")
    print("The title is: " , driver.title)
    wait = WebDriverWait(driver, 10)
    driver.find_element(By.CSS_SELECTOR, "#ajax-content > button").click()
    wait.until(expected_conditions.text_to_be_present_in_element((By.TAG_NAME, "h1"),"HELLO!"))
    hello_text = driver.find_element(By.TAG_NAME, "h1").text
    print("The text is: ", hello_text)
    wait.until(expected_conditions.text_to_be_present_in_element((By.TAG_NAME, "h3"),"I'm late!"))
    late_text = driver.find_element(By.TAG_NAME, "h3").text
    print("The text is: ", late_text)
